xarray/tests/test_concat.py

```diff
@@ -46,14 +145,612 @@ def test_concat_compat() -> None:
 
     for var in ["has_x", "no_x_y"]:
         assert "y" not in result[var].dims and "y" not in result[var].coords
-    with pytest.raises(
-        ValueError, match=r"coordinates in some datasets but not others"
-    ):
+    with pytest.raises(ValueError, match=r"'q' not present in all datasets"):
         concat([ds1, ds2], dim="q")
-    with pytest.raises(ValueError, match=r"'q' is not present in all datasets"):
+    with pytest.raises(ValueError, match=r"'q' not present in all datasets"):
         concat([ds2, ds1], dim="q")
 
 
+def test_concat_missing_var() -> None:
+    datasets = create_concat_datasets(2, 123)
+    vars_to_drop = ["humidity", "precipitation", "cloud cover"]
```

Comment:
Should be cleaner to construct the expected result by concatenating before dropping variables
```suggestion
    expected = xr.concat(datasets, dim="day")
    vars_to_drop = ["humidity", "precipitation", "cloud cover"]
```

So we can avoid the manual concatenation below

Reply:
I've adapted the tests, looking much cleaner now.